src/main/java/atrium/AtriumClient.java

```diff
@@ -0,0 +1,526 @@
+    this.mxAPIKEY = mxAPIKEY;
+    this.mxCLIENTID = mxCLIENTID;
+  }
+
+
+  // USER
+
+
+  // Required Parameters: None
+  // Optional Parameters: identifier, isDisabled, metadata
+  public User createUser(String identifier, String isDisabled, String metadata) {
+    JsonObject inner = new JsonObject();
+    if (!identifier.equals("")) {
+      inner.addProperty("identifier", identifier);
+    }
+    isDisabled = isDisabled.toLowerCase();
+    if ((isDisabled.equals("")) || (isDisabled.equals("false"))) {
+      inner.addProperty("is_disabled", "false");
+    }
+    else {
+      inner.addProperty("is_disabled", true);
+    }
+    if (!metadata.equals("")) {
+      inner.addProperty("metadata", metadata);
+    }
+    JsonObject outer = new JsonObject();
+    outer.add("user", inner);
+    String body = outer.toString();
+
+    String response = makeRequest("POST", "/users", body);
+    return new Gson().fromJson(new Gson().fromJson(response, JsonObject.class).get("user").toString(), User.class);
+  }
+
+  // Required Parameters: userGUID
+  // Optional Parameters: None
+  public User readUser(String userGUID) {
+    String response =  makeRequest("GET", "/users/" + userGUID, "");
+    return new Gson().fromJson(new Gson().fromJson(response, JsonObject.class).get("user").toString(), User.class);
+  }
+
+  // Required Parameters: None
+  // Optional Parameters: userGUID, identifier, isDisabled, metadata
+  public User updateUser(String userGUID, String identifier, String isDisabled, String metadata) {
+    JsonObject inner = new JsonObject();
+    if (!identifier.equals("")) {
+      inner.addProperty("identifier", identifier);
+    }
+    if (!isDisabled.equals("")) {
+      inner.addProperty("is_disabled", isDisabled);
+    }
+    if (!metadata.equals("")) {
+      inner.addProperty("metadata", metadata);
+    }
+    JsonObject outer = new JsonObject();
+    outer.add("user", inner);
+    String body = outer.toString();
+
+    String response = makeRequest("PUT", "/users/" + userGUID, body);
+    return new Gson().fromJson(new Gson().fromJson(response, JsonObject.class).get("user").toString(), User.class);
+  }
+
+  // Required Parameters: None
+  // Optional Parameters: pageNumber, recordsPerPage
+  public User[] listUsers(String pageNumber, String recordsPerPage) {
+    String params = optionalParameters("", "", "", pageNumber, recordsPerPage);
+    String response = makeRequest("GET", "/users" + params, "");
+    JsonParser parser = new JsonParser();
+    JsonElement userList = parser.parse(new Gson().fromJson(response, JsonObject.class).get("users").toString());
+    JsonArray jsonArray = userList.getAsJsonArray();
+    User[] userArray = new User[jsonArray.size()];
+    for (int i = 0; i < jsonArray.size(); i++) {
+      userArray[i] = new Gson().fromJson(jsonArray.get(i), User.class);
+    }
+    return userArray;
+  }
+
+  // Required Parameters: userGUID
+  // Optional Parameters: None
+  public String deleteUser(String userGUID) {
```

Comment:
Not the biggest deal for an example, but its best practice to always validate params being sent, and bail if a required param isn't sent.